openpype/pype_commands.py

```diff
@@ -104,6 +111,122 @@ def publish(paths, targets=None):
         log.info("Publish finished.")
         uninstall()
 
+    @staticmethod
+    def remotepublish(project, batch_path, host, user, targets=None):
+        """Start headless publishing.
+
+        Publish use json from passed paths argument.
+
+        Args:
+            project (str): project to publish (only single context is expected
+                per call of remotepublish
+            batch_path (str): Path batch folder. Contains subfolders with
+                resources (workfile, another subfolder 'renders' etc.)
+            targets (string): What module should be targeted
+                (to choose validator for example)
+            host (string)
+            user (string): email address for webpublisher
+
+        Raises:
+            RuntimeError: When there is no path to process.
+        """
+        if not batch_path:
+            raise RuntimeError("No publish paths specified")
+
+        from openpype import install, uninstall
+        from openpype.api import Logger
+        from openpype.lib import OpenPypeMongoConnection
+
+        # Register target and host
+        import pyblish.api
+        import pyblish.util
+
+        log = Logger.get_logger()
+
+        log.info("remotepublish command")
+
+        install()
+
+        if host:
+            pyblish.api.register_host(host)
+
+        if targets:
+            if isinstance(targets, str):
+                targets = [targets]
+            for target in targets:
+                pyblish.api.register_target(target)
+
+        os.environ["OPENPYPE_PUBLISH_DATA"] = batch_path
+        os.environ["AVALON_PROJECT"] = project
+        os.environ["AVALON_APP"] = host  # to trigger proper plugings
+
+        # this should be more generic
+        from openpype.hosts.webpublisher.api import install as w_install
+        w_install()
+
+        log.info("Running publish ...")
+
+        # Error exit as soon as any error occurs.
+        error_format = "Failed {plugin.__name__}: {error} -- {error.traceback}"
+
+        mongo_client = OpenPypeMongoConnection.get_mongo_client()
+        database_name = os.environ["OPENPYPE_DATABASE_NAME"]
+        dbcon = mongo_client[database_name]["webpublishes"]
+
+        _, batch_id = os.path.split(batch_path)
+        _id = dbcon.insert_one({
```

Comment:
What if document with `batch_id` already exists?

Reply:
There could be multiple documents with same `batch_id`, when batch fails, it might get reprocessed by admin, new document is created with same `batch_id`. This is why `_id` is collected to update document and not only `batch_id`.